test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,246 @@
+        self.assertConfigureFails(PitchContours(), {'peakFrameThreshold': -1})
+        self.assertConfigureFails(PitchContours(), {'pitchContinuity': -1})
+        self.assertConfigureFails(PitchContours(), {'sampleRate': -1})
+        self.assertConfigureFails(PitchContours(), {'timeContinuity': -1})
+
+    def testDuration(self):
+        # simple test for the duration output with small populated frames
+        nonEmptyPeakBins = testPeakBins
+        nonEmptyPeakSaliences = testPeakSaliences
+        theHopSize= 2 * defaultHopSize
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(nonEmptyPeakBins, nonEmptyPeakSaliences)
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+        
+        theHopSize= 4 * defaultHopSize
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(nonEmptyPeakBins, nonEmptyPeakSaliences)
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+        theHopSize= 8 * defaultHopSize
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(nonEmptyPeakBins, nonEmptyPeakSaliences)
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)                       
+    
+    def testEmpty(self):
+        emptyPeakBins = []
+        emptyPeakSaliences = []
+        bins, saliences, startTimes, duration = PitchContours()(emptyPeakBins, emptyPeakSaliences)
+
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        calculatedDuration = (2*defaultHopSize)/defaultSampleRate
+        self.assertEqual(duration, 0)
+
+    def testEmptyColumns(self):
+        emptyPeakBins = [[],[]]
+        emptyPeakSaliences = [[],[]]
+        theHopSize= 2*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        calculatedDuration = (2*theHopSize)/defaultSampleRate
+        # TODO This is a huge relative error threshold. Why is that?
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    # FIXME - This test case is failing
+    # The test is set up so that random frequency peaks are fed into PitchSalienceFunction, 
+    # chained to PitchSalienceFunctionPeaks and then fed to PitchContours.
+    # The end result produces empty output.
+    def testRandomPeaks(self):
+        freq_speaks = []
+        mag_speaks = []
+
+        # Randomize peak inputs
+        index = 0
+        while index < 4:
+            f1 = random.randrange(100, 500)
+            freq_speaks.append(f1) 
+            mag_speaks.append(1) 
+            index+=1
```

Comment:
Here, you are selecting a few random frequencies, and pass them to  `PitchSalienceFunction`. This is hard to evaluate because the values are random. Instead you could pass a single frequency (each one resulting in a straight line pitch contour, and evaluate a few of them):
- 55 Hz (corresponds to 0th bin in pitch salience function)
- 440Hz